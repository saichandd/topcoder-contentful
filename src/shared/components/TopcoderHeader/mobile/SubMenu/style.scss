@import '~styles/mixins';

.item {
  color: $tc-white;
  font: 500 14px "Sofia Pro", sans-serif;
  margin: 9px 18px;
  text-transform: uppercase;

  a:hover {
    color: $tc-dark-blue;
  }
}

.sub-menu {
  padding: 6px 0;
}

.title {
  color: $tc-gray-40;
  font: 500 18px "Sofia Pro", sans-serif;
  margin: 9px 18px;
  text-transform: uppercase;
}
