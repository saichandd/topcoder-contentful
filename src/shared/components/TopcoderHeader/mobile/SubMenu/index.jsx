import PT from 'prop-types';
import React from 'react';
import './style.scss';

export default function SubMenu({
  hideTitle,
  subMenu,
}) {
  const items = subMenu.items.map(item => (
    <li key={item.title} styleName="item">
      <a href={item.link}>
        {item.title}
      </a>
    </li>
  ));
  const title = hideTitle ? '' : (
    <li styleName="title">
      {subMenu.title}
    </li>
  );
  return (
    <ul styleName="sub-menu">
      {title}
      {items}
    </ul>
  );
}

export const SUB_MENU_SHAPE = PT.shape({
  title: PT.string.isRequired,
  items: PT.arrayOf(PT.shape({
    icon: PT.node.isRequired,
    link: PT.string.isRequired,
    title: PT.string.isRequired,
  })).isRequired,
});

SubMenu.defaultProps = {
  hideTitle: false,
};

SubMenu.propTypes = {
  hideTitle: PT.bool,
  subMenu: SUB_MENU_SHAPE.isRequired,
};
