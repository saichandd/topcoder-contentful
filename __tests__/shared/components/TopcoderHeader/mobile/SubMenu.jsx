import React from 'react';
import R from 'react-test-renderer/shallow';
import SubMenu from 'components/TopcoderHeader/mobile/SubMenu';

const renderer = new R();

test('Matches snapshot', () => {
  renderer.render((
    <SubMenu
      subMenu={{
        title: 'Menu title',
        items: [{
          icon: <div />,
          link: '/LINK',
          title: 'Item title',
        }],
      }}
    />
  ));
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
