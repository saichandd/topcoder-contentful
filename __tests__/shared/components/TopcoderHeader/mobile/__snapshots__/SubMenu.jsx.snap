// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<ul
  className="src-shared-components-TopcoderHeader-mobile-SubMenu-___style__sub-menu___TEWxN"
>
  <li
    className="src-shared-components-TopcoderHeader-mobile-SubMenu-___style__title___3DKHU"
  >
    Menu title
  </li>
  <li
    className="src-shared-components-TopcoderHeader-mobile-SubMenu-___style__item___2omud"
  >
    <a
      href="/LINK"
    >
      Item title
    </a>
  </li>
</ul>
`;
